src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1608,6 +1964,58 @@ void EIRPlantLoopHeatPump::processInputForEIRPLHP(EnergyPlusData &state)
+                                                                                           thisPLHP.name,
+                                                                                           DataLoopNode::NodeFluidType::Water,
+                                                                                           DataLoopNode::ConnectionType::Inlet,
+                                                                                           NodeInputManager::CompFluidStream::Tertiary,
+                                                                                           DataLoopNode::ObjectIsNotParent);
+                    thisPLHP.heatRecoveryNodes.outlet = NodeInputManager::GetOnlySingleNode(state,
+                                                                                            heatRecoveryOutletNodeName,
+                                                                                            nodeErrorsFound,
+                                                                                            objType,
+                                                                                            thisPLHP.name,
+                                                                                            DataLoopNode::NodeFluidType::Water,
+                                                                                            DataLoopNode::ConnectionType::Outlet,
+                                                                                            NodeInputManager::CompFluidStream::Tertiary,
+                                                                                            DataLoopNode::ObjectIsNotParent);
+
+                    thisPLHP.heatRecoveryDesignVolFlowRate =
+                        state.dataInputProcessing->inputProcessor->getRealFieldValue(fields, schemaProps, "heat_recovery_reference_flow_rate");
+                    if (thisPLHP.heatRecoveryDesignVolFlowRate == DataSizing::AutoSize) {
+                        thisPLHP.heatRecoveryDesignVolFlowRateWasAutoSized = true;
+                    }
+
+                    // fields only in cooling object
+                    if (thisPLHP.heatRecoveryAvailable && thisPLHP.EIRHPType == DataPlant::PlantEquipmentType::HeatPumpEIRCooling) {
+                        thisPLHP.maxHeatRecoveryTempLimit = state.dataInputProcessing->inputProcessor->getRealFieldValue(
+                            fields, schemaProps, "maximum_heat_recovery_outlet_temperature");
+                    }
+                    // fields only in heating object
+                    if (thisPLHP.heatRecoveryAvailable && thisPLHP.EIRHPType == DataPlant::PlantEquipmentType::HeatPumpEIRHeating) {
+                        thisPLHP.minHeatRecoveryTempLimit = state.dataInputProcessing->inputProcessor->getRealFieldValue(
+                            fields, schemaProps, "minimum_heat_recovery_outlet_temperature");
+                    }
+                }
+
```

Comment:
If heat recovery is available and the HP's condenser type is `airsource`, then get the actual inlet and outlet nodes and process the remaining heat recovery user input fields.